flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -123,6 +130,32 @@ private FailureHandlingResult handleFailure(
                     globalFailure);
         }
 
+        if (!globalFailure) {
+            try {
+                final SchedulingExecutionVertex failedVertex =
+                        schedulingTopology.getVertex(failingExecutionVertexId);
+                if (cause instanceof PartitionException
+                        && !failedVertex.getCacheIntermediateDataSetIds().isEmpty()) {
+                    return FailureHandlingResult.unrecoverable(
+                            failingExecutionVertexId,
+                            new CacheCorruptedException(
+                                    "Fail to consume cache",
+                                    cause,
+                                    failedVertex.getCacheIntermediateDataSetIds()),
+                            timestamp,
+                            globalFailure);
+                } else if (cause instanceof CacheCorruptedException) {
```

Comment:
This translation can be avoided by annotating `CacheCorruptedException` with `ThrowableType#NonRecoverableError`.